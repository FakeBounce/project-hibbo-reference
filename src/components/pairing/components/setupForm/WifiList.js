import React, { PureComponent } from 'react';
import { Animated } from 'react-native';
import { func, shape, string, instanceOf, bool } from 'prop-types';
import styled from 'styled-components/native';
import Config from 'react-native-config';

import appStyles from 'styles/appStyles';

import { StyledContainerBasic } from 'styledComponents/containers';

import Title from '../Title';
import WifiName from './WifiName';

const { IS_BLE_ON } = Config;
const SScrollView = styled.ScrollView`
  flex: 1;
  background-color: transparent;
`;
export const AnimatedScrollView = styled(
  Animated.createAnimatedComponent(SScrollView),
)``;

class WifiList extends PureComponent {
  componentDidMount() {
    if (IS_BLE_ON === 'true') {
      this.props.initBle();
    }
  }

  render() {
    const {
      title,
      setData,
      titleAnimations,
      wifiListAnimations,
      showContent,
      connected,
      scanning,
      scanStatus,
      SSIDs,
    } = this.props;

    return (
      <StyledContainerBasic>
        <Title
          label={[title.label1, title.label2, title.label3]}
          color={appStyles.colors.white}
          animations={titleAnimations}
          showContent={showContent}
        />

        <AnimatedScrollView
          style={{
            opacity: wifiListAnimations.opacity,
            transform: [{ translateY: wifiListAnimations.translateY }],
          }}
        >
          {!connected && <WifiName label="Non connectÃ©" action={setData} />}
          {connected &&
            scanning && <WifiName label="Scanning" action={() => {}} />}
          {connected &&
            scanStatus === 'success' &&
            Array.from(SSIDs.keys()).map((value, index) => {
              return (
                <WifiName
                  key={`${value}${index}`}
                  label={value}
                  action={setData}
                />
              );
            })}
        </AnimatedScrollView>
      </StyledContainerBasic>
    );
  }
}

WifiList.propTypes = {
  setData: func.isRequired,
  title: shape({
    label1: string,
    label2: string,
    label3: string,
  }).isRequired,
  titleAnimations: shape({
    opacity: instanceOf(Animated.Value).isRequired,
    translateY: instanceOf(Animated.Value).isRequired,
    scaleY: instanceOf(Animated.Value).isRequired,
  }).isRequired,
  wifiListAnimations: shape({
    opacity: instanceOf(Animated.Value).isRequired,
    translateY: instanceOf(Animated.Value).isRequired,
  }).isRequired,
  showContent: func.isRequired,
  connected: bool.isRequired,
  scanning: bool.isRequired,
  scanStatus: string.isRequired,
  SSIDs: shape({
    ssid: string,
  }).isRequired,
  initBle: func.isRequired,
};

export default WifiList;
