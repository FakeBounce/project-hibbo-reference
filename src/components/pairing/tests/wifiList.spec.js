import React from 'react';
import renderer from 'react-test-renderer';

import WifiList from '../components/setupForm/WifiList';

describe('WifiList', () => {
  const fn = jest.fn();
  const component = renderer.create(
    <WifiList
      setData={fn}
    />,
  ).toJSON();

  it('should render the component', () => {
    expect(component).toMatchSnapshot();
  });
});
