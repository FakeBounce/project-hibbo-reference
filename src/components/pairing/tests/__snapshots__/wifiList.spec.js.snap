// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WifiList should render the component 1`] = `
<RCTScrollView
  style={
    Array [
      undefined,
    ]
  }
>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        Wifi 1
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        Wifi 2
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        Wifi 3
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={undefined}
      >
        Wifi 4
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
